src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -545,7 +578,24 @@ public function register(Container $app)
             };
         });
 
-        $app['security.authentication_provider.dao._proto'] = $app->protect(function ($name) use ($app) {
+        $app['security.entry_point.guard._proto'] = $app->protect(function ($name, array $options) use ($app) {
+            if (isset($options['entry_point'])) {
+                // if it's configured explicitly, use it!
+                return $app[$options['entry_point']];
+            }
+            $authenticatorIds = $options['authenticators'];
+            if (count($authenticatorIds) == 1) {
+                // if there is only one authenticator, use that as the entry point
+                return $app[array_shift($authenticatorIds)];
```

Comment:
using array_shift to get the first argument creates a reference mismatch here on PHP 5.x (PHP 7 does not have reference mismatches anymore) because the function takes a reference as argument. This forces to duplicate the array.
PHP 7 will also need to duplicate the array as you alter it.

Getting the first element without changing the array would be better.